### PR TITLE
[IMP] developer: improve section on env

### PR DESCRIPTION
**Pull request by ged-odoo**
- add a link to services
- add a note that the env is quite powerful
- reformat the table with the list-table role, to make it easier to edit

### CODE REVIEW

File: content/developer/reference/javascript/framework_overview.rst [hunk 1]
```diff
@@ -99,22 +99,25 @@ As an Owl application, the Odoo web client defines its own environment (componen
...
+     - :ref:`main bus <javascript/bus>`, used to coordinate some generic events
+   * - `services`
+     - all deployed :ref:`services <frontend/services>` (should usually be accessed
+       with the `useService` hook)
+   * - `debug`
+     - boolean. If true, the web client is in `debug` mode
+   * - `_t`
+     - translation function
+   * - `isSmall`
+     - boolean. If true, the web client is currently in mobile mode
```
Comment (aab-odoo) [inline content/developer/reference/javascript/framework_overview.rst:120]: screen width <= 767px is more accurate than just "mobile mode"

File: content/developer/reference/javascript/framework_overview.rst [hunk 2]
```diff
@@ -125,6 +128,12 @@ can do this:
 
     const someString = this.env._t('some text');
 
+.. note::
+
+   Having a reference to the environment is quite powerful, because it provides
+   access to all services. This is quite useful in many cases. For example,
```
Comment (aab-odoo) [inline content/developer/reference/javascript/framework_overview.rst:134]: I would remove at least one of the two "quite"

**Review by aab-odoo — APPROVED** [review]

**Comment by ged-odoo** [comment]
a pr for @odoo/doc-review

**Comment by ged-odoo** [comment]
just rebased

**Review by Feyensv — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**